File: scripts/deployment/render-env-importer.js
#!/usr/bin/env node

/**
 * Render Environment Variables Importer
 * 
 * This script helps you quickly copy environment variables to Render dashboard.
 * It reads the render-config.json file and formats the variables for easy copying.
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

console.log('🚀 Render Environment Variables Importer');
console.log('==========================================\n');

try {
    // Read the configuration file
    const configPath = path.join(__dirname, 'render-config.json');
    const configData = fs.readFileSync(configPath, 'utf8');
    const config = JSON.parse(configData);

    const { envVars, sensitiveVars, instructions } = config.render;

    console.log('📋 INSTRUCTIONS:');
    Object.entries(instructions).forEach(([key, instruction]) => {
        console.log(`   ${key}. ${instruction}`);
    });
    console.log('');

    console.log('🔧 NON-SENSITIVE ENVIRONMENT VARIABLES:');
    console.log('   (Copy these directly to Render dashboard)\n');
    
    Object.entries(envVars).forEach(([key, value]) => {
        console.log(`${key}=${value}`);
    });

    console.log('\n🔐 SENSITIVE VARIABLES (SET MANUALLY):');
    console.log('   (Replace with your actual values)\n');

    console.log('   REQUIRED:');
    sensitiveVars.required.forEach(varName => {
        console.log(`   ${varName}=YOUR_ACTUAL_VALUE_HERE`);
    });

    console.log('\n   OPTIONAL:');
    sensitiveVars.optional.forEach(varName => {
        console.log(`   ${varName}=YOUR_ACTUAL_VALUE_HERE`);
    });

    console.log('\n📝 QUICK COPY FORMAT:');
    console.log('   Use this format in Render dashboard:\n');
    
    // Show first few as examples
    const examples = Object.entries(envVars).slice(0, 3);
    examples.forEach(([key, value]) => {
        console.log(`   Key: ${key}`);
        console.log(`   Value: ${value}`);
        console.log('   ---');
    });

    console.log('\n✅ NEXT STEPS:');
    console.log('   1. Generate APP_KEY: php artisan key:generate --show');
    console.log('   2. Get database credentials from your provider');
    console.log('   3. Go to Render dashboard > Your Service > Environment');
    console.log('   4. Add each variable using "Add Environment Variable" button');
    console.log('   5. Deploy your service');

    console.log('\n🔗 HEALTH CHECK:');
    console.log('   After deployment, check: https://your-app-name.onrender.com/debug/health');

} catch (error) {
    console.error('❌ Error reading configuration:', error.message);
    console.log('\n💡 Make sure render-config.json exists in the same directory.');
}

console.log('\n==========================================');
console.log('🎉 Configuration export completed!');

File: scripts/deployment/render-env-importer.py
#!/usr/bin/env python3
"""
Render Environment Variables Importer (Python Version)

This script helps you quickly format environment variables for Render dashboard.
It reads the render-config.json file and provides formatted output.
"""

import json
import os

def main():
    print("🚀 Render Environment Variables Importer (Python)")
    print("=" * 50)
    print()

    try:
        # Read the configuration file
        config_path = os.path.join(os.path.dirname(__file__), 'render-config.json')
        
        with open(config_path, 'r') as f:
            config = json.load(f)

        render_config = config['render']
        env_vars = render_config['envVars']
        sensitive_vars = render_config['sensitiveVars']
        instructions = render_config['instructions']

        print("📋 INSTRUCTIONS:")
        for key, instruction in instructions.items():
            print(f"   {key}. {instruction}")
        print()

        print("🔧 NON-SENSITIVE ENVIRONMENT VARIABLES:")
        print("   (Copy these directly to Render dashboard)")
        print()
        
        for key, value in env_vars.items():
            print(f"{key}={value}")

        print()
        print("🔐 SENSITIVE VARIABLES (SET MANUALLY):")
        print("   (Replace with your actual values)")
        print()

        print("   REQUIRED:")
        for var_name in sensitive_vars['required']:
            print(f"   {var_name}=YOUR_ACTUAL_VALUE_HERE")

        print()
        print("   OPTIONAL:")
        for var_name in sensitive_vars['optional']:
            print(f"   {var_name}=YOUR_ACTUAL_VALUE_HERE")

        print()
        print("📝 RENDER DASHBOARD FORMAT:")
        print("   Add variables one by one in this format:")
        print()
        
        # Show first few as examples
        example_vars = list(env_vars.items())[:3]
        for key, value in example_vars:
            print(f"   Variable Name: {key}")
            print(f"   Variable Value: {value}")
            print("   " + "-" * 30)

        print()
        print("✅ NEXT STEPS:")
        print("   1. Generate APP_KEY: php artisan key:generate --show")
        print("   2. Get database credentials from your provider")
        print("   3. Go to Render dashboard > Your Service > Environment")
        print("   4. Click 'Add Environment Variable' for each variable")
        print("   5. Deploy your service")

        print()
        print("🔗 POST-DEPLOYMENT:")
        print("   Health Check: https://your-app-name.onrender.com/debug/health")
        print("   App Info: https://your-app-name.onrender.com/debug/info")

        print()
        print("💡 TIP: You can also use the render.yaml file for automatic deployment!")

    except FileNotFoundError:
        print("❌ Error: render-config.json not found!")
        print("   Make sure you're running this script from the project directory.")
    except json.JSONDecodeError:
        print("❌ Error: Invalid JSON in render-config.json!")
    except Exception as e:
        print(f"❌ Error: {str(e)}")

    print()
    print("=" * 50)
    print("🎉 Configuration export completed!")

if __name__ == "__main__":
    main()
